src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/AutoInsert/OnAutoInsertEndpointTest.cs

```diff
@@ -239,6 +276,152 @@ public async Task Handle_NoApplicableProvider_CallsProviderAndReturnsNull()
             // Assert
             Assert.Null(result);
             Assert.True(insertProvider.Called);
+            Assert.Equal(0, LanguageServer.RequestCount);
+        }
+
+        [Fact]
+        public async Task Handle_SingleServer_CSharpDocCommentSnippet()
+        {
+            // Arrange
+            var input = """
+                <div>
+                </div>
+
+                @functions {
+                    ///$$
+                    public void M()
+                    {
+                    }
+                }
+                """;
+
+            var expected = """
+                <div>
+                </div>
+
+                @functions {
+                    /// <summary>
+                    /// $0
+                    /// </summary>
+                    public void M()
+                    {
+                    }
+                }
+                """;
+
+            var character = "/";
+
+            await VerifyCSharpOnAutoInsertAsync(input, expected, character);
+        }
+
+        [Fact]
+        public async Task Handle_SingleServer_CSharpDocCommentNewLine()
+        {
+            // Arrange
+            var input = """
+                <div>
+                </div>
+
+                @functions {
+                    /// <summary>
+                    /// This is some text
+                    $$
+                    /// </summary>
+                    public void M()
+                    {
+                    }
+                }
+                """;
+
+            var expected = """
+                <div>
+                </div>
+
+                @functions {
+                    /// <summary>
+                    /// This is some text
+                    /// $0
+                    /// </summary>
+                    public void M()
+                    {
+                    }
+                }
+                """;
+
+            var character = "\n";
+
+            await VerifyCSharpOnAutoInsertAsync(input, expected, character);
+        }
+
+        [Fact(Skip = "Roslyn only responds to the Razor server kind for this request, but uses the C# server kind in tests")]
```

Comment:
Is there a bug tracking this? When first implementing the C# LSP test server I thought I set it up with the Razor server kind, but looks like it's buggy 🤔

Reply:
No, but good idea. I don't think this is a bug, its just something that you didn't expose a way to set in EA (or was really hard to plumb through).

Also, in case I haven't said it enough, thank you 10000 times for allowing us to use the real C# LSP server, because it gives me _so_ much more confidence in our tests that I've ever had.